dotnet/src/Functions/Functions.Prompty.UnitTests/PromptyTest.cs

```diff
@@ -151,6 +181,36 @@ public async Task ItSupportsSeparatorInContentAsync()
             """, await kernelFunction.InvokeAsync<string>(kernel));
     }
 
+    [Theory]
+    [InlineData("""
+        ---
+        name: MyPrompt
+        ---
+        {% for item in items %}
+        {{a}} {{b}} {{c.d}} {{e.f.g}} {{a}} {{c}} {{f}}
+        {% endfor %}
+        """, new string[] { "a", "b" })]
+    [InlineData("""
+        ---
+        name: MyPrompt
+        inputs:
+          question: What is the color of the sky?
+        ---
+        {{a}} {{b}} {{c}}
+        """, new string[] { "question" })]
+    public void ItCreatesInputVariablesForThoseImplicitlyFoundIfNoneAreExplicit(string prompty, string[] expectedVariables)
```

Comment:
I think this would be clearer as two seperate tests, what do you think?

1. `ItCreatesInputVariablesOnlyWhenNoneAreExplicitlySet`
2. `ItCreatesInputVariablesOnlyForSimpleVariables`

Reply:
Personally I prefer the theory approach, but either way is fine, and I changed it.